### PR TITLE
119814 modal displays 0

### PR DESCRIPTION
**Pull request by numbap**
## [ADO-119814](https://dev.azure.com/VP-BD/DECD/_workitems/edit/119814)

Fixed [AB#119814](https://dev.azure.com/VP-BD/d5ca3cd4-19cb-4c8c-b3e6-a7068f4677e4/_workitems/edit/119814)

### Description

List of proposed changes:

- Ensured that modal will not pop up unless the user has been inactive for a minimum period of time. 
- Inactivity is defined as mouse movements (PC) and scrolls (Mobile device)

### What to test for/How to test

 - Go to my-dashboard and jiggle the mouse for one minute. Nothing should happen. 
 - Leave idle for 30 seconds, timer modal should appear
 - Switch to mobile interface, scroll up and down for one minute. Nothing should happen. 
 - Leave idle on mobile interface for 1 minute. Modal should appear. 
 - When modal appears, click Signout button. User should be redirected. 
 - When modal appears, click on "Stay signed in" and repeat previous steps. Modal should act as expected.

### CODE REVIEW

File: components/sessionModals/SessionTimeTracker.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          }
+        })
+      }
+    }
+    window.addEventListener('mousemove', updateModalTimer)
+    window.addEventListener('scroll', updateModalTimer)
+    return () => {
+      window.removeEventListener('mousemove', updateModalTimer)
+      window.removeEventListener('scroll', updateModalTimer)
+    }
```
Comment (krischarbonneau) [inline components/sessionModals/SessionTimeTracker.js:46]: While this is okay, I feel there's a lot more events we can track to determine if a user is idle. See the default props for `react-idle-timer` here[](https://idletimer.dev/docs/api/props#events) as an example.

**Review by krischarbonneau — COMMENTED** [review]
While this works as intended, I think we can clean up our implementation a lot. Ideally we should reuse the Modal component rather than creating another modal component. If react-modal doesn't provide exactly what we need we can create our own modal in its place and have optional props to suit our needs rather than having multiple modal components.

I'd recommend using `react-idle-timer` to control activity detection for the countdown functionality. Just some thoughts, open to discussion.

File: components/sessionModals/SessionTimeTracker.js (hunk 1 above)
Comment (numbap) [reply components/sessionModals/SessionTimeTracker.js]: I've implemented react-idle-timer, and will now focus on the modal comment you'd made.

File: components/sessionModals/SessionTimeTracker.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  const onStay = () => {
+    reset()
+    setDemoModalBody(null)
+  }
+
+  const tick = useCallback(() => {
+    const minutes = Math.floor(getRemainingTime() / 60000)
+    const seconds = Math.floor((getRemainingTime() / 1000) % 60).toFixed(0)
+    console.log(getRemainingTime(), isPrompted())
```
Comment (krischarbonneau) [inline components/sessionModals/SessionTimeTracker.js:28]: Remove this :)

File: components/sessionModals/SessionTimeTracker.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  const [demoModalBody, setDemoModalBody] = useState(null)
+
+  const handleOnIdle = () => {
+    Router.push('/auth/logout')
+  }
+
+  const { isPrompted, reset, getRemainingTime } = useIdleTimer({
+    onIdle: handleOnIdle,
+    promptBeforeIdle: 0.5 * 60 * 1000, // 30 seconds
+    timeout: 1 * 60 * 1000, // 1 minute
```
Comment (krischarbonneau) [inline components/sessionModals/SessionTimeTracker.js:17]: This amount of time could be bad UX. I think the decision should ultimately be up to Farwa but I'd have the timeout set to 10 minutes and the prompt before logging the user out to be 5 minutes.

File: package.json [hunk 4]
```diff
@@ -33,6 +33,7 @@
     "@opentelemetry/api": "^1.3.0",
     "@opentelemetry/auto-instrumentations-node": "^0.36.0",
     "@opentelemetry/sdk-node": "^0.35.0",
+    "@reach/auto-id": "^0.18.0",
```
Comment (krischarbonneau) [inline package.json:36]: accidental install?

File: components/sessionModals/CountDown.js [hunk 5]
```diff
@@ -71,8 +41,9 @@ const Countdown = (props) => {
         <div className="flex-auto">
           <p className="mr-6">{props.bannerContent[0]}</p>
           <p className="font-bold mr-6">
-            {ready &&
-              `${props.bannerContent[1]} ${clock.minutes} ${props.bannerMinutesAnd} ${clock.seconds} ${props.bannerSeconds}.`}
+            {props.minutes || props.seconds
+              ? `${props.bannerContent[1]} ${props.xminutes} ${props.bannerMinutesAnd} ${props.xseconds} ${props.bannerSeconds}.`
```
Comment (krischarbonneau) [inline components/sessionModals/CountDown.js:45]:
```suggestion
              ? `${props.bannerContent[1]} ${props.minutes} ${props.bannerMinutesAnd} ${props.seconds} ${props.bannerSeconds}.`
```
Looks like `xminutes` should just be `minutes` based on `Countdown.js`, same with seconds :)

File: pages/my-dashboard.js [hunk 6]
```diff
@@ -162,14 +118,10 @@ export default function MyDashboard(props) {
...
       </Modal>
-      <Modal
-        className="flex justify-center bg-black/75 h-full"
-        isOpen={demoModalBody === null ? false : true}
-        onRequestClose={closeModal}
-        contentLabel={'Demo Modal'}
-      >
-        {demoModalBody}
-      </Modal>
+      <SessionTimeTracker
```
Comment (krischarbonneau) [inline pages/my-dashboard.js:121]: Just a nitpick so you can ignore this if you want, but would a better name for this component be `SessionTimeout`?

**Review by krischarbonneau — APPROVED** [review]
Good job! LGTM

**Pull request merged**